run.py

```diff
@@ -14,4 +16,40 @@
     
 toolbar = DebugToolbarExtension(app)
 
-connect_db(app)
\ No newline at end of file
+connect_db(app)
+
+########################################################
+# User Signup/Login/Logout Routes:
+
+@app.route('/signup', methods=['GET', 'POST'])
+def signup():
+    """Handle user signup"""
+
+    form = UserAddForm()
+
+    if form.validate_on_submit():
+        # Attempt to create a user
+
+        errors = handle_signup_errors(form.username.data, form.email.data, None)
+        if errors:
+            return render_template('users/signup.html', form=form)
+
+        try:
+            User.signup(
+                username=form.username.data,
+                password=form.password.data,
+                email=form.email.data,
+                first_name=form.first_name.data,
+                last_name=form.last_name.data,
+                image_url=form.image_url.data or User.image_url.default.arg
+            )
+            db.session.commit()
+
+            return redirect('/signup/favorites')
+        # If db errors, redirect back to try again
+        except exc.SQLAlchemyError as e:
+            print(f'ERROR AT LINE 51. ERROR TYPE: {type(e)}')
```

Comment:
@iTaylorCodes what does `ERROR AT LINE 51` mean ?

Reply:
@alexkayabula Was printing error to server log, and giving it a location the error took place. I'll fix that 👍